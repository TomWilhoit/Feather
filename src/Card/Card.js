import React from 'react'
import './Card.scss'
import PropTypes from "prop-types"
import { connect } from "react-redux";

export const Card = (props) => {
  if(props.weather.length){
    const temperature = props.weather[props.place.id].currently.apparentTemperature
    const summary = props.weather[props.place.id].daily.summary
  return (
    <a onClick={props.toggleDisplay}  value={props.place.id} className='place-card'>
      <h2>{props.place.ficName}</h2>
      <h1>{temperature} F</h1>
      <h4>{summary}</h4>
      <h5>Inspiration: {props.place.realName}</h5>
    </a>
  )
  }else{
    return(<div>Loading</div>)
  }
}

Card.propTypes = {
  locations: PropTypes.array,
  weather: PropTypes.array
};

export const mapStateToProps = state => ({
  locations: state.locations,
  weather: state.weather
});

export default connect(
  mapStateToProps,
  null
)(Card);