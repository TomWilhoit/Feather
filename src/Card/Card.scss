.place-card {
  width: 400px;
    background: darkseagreen;
    color: white;
    text-decoration: none;
    display: flex;
    flex-direction: column;
    height: 300px;
    border-radius: 5px;
    margin: 20px;
}