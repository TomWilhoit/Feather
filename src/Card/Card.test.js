import React from "react";
import { Card } from "./Card";
import { mapStateToProps } from "./Card";
import { shallow } from "enzyme";

describe("Card", () => {
  let wrapper;
  const mockProps = {
    place: {
      id: 0,
      realName: "Chattanooga",
      ficName: "Chattavegas",
      reason: "I know. It is so dumb"
    },
    weather: [
      {
        latitude: 40.7128,
        longitude: -74.006,
        timezone: "America/New_York",
        currently: {
          time: 1555470242,
          summary: "Overcast",
          apparentTemperature: 56.61,
          
        },
        daily: {summary: 'It is raining sideways'}
      }
    ]
  };
  beforeEach(() => {
    wrapper = shallow(<Card {...mockProps} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with all data passed in", () => {
    const mockProps = {weather: []};
    wrapper = shallow(<Card {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    it("should mapStateToProps", () => {
      const mockState = {
        locations: ["Chattanooga", "Memphis"]
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
    });
  });
});
