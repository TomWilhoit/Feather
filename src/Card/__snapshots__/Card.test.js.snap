// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in 1`] = `
<a
  className="place-card"
  value={0}
>
  <h2>
    Chattavegas
  </h2>
  <h1>
    56.61
     F
  </h1>
  <h4>
    It is raining sideways
  </h4>
  <h5>
    Inspiration: 
    Chattanooga
  </h5>
</a>
`;

exports[`Card should match the snapshot with all data passed in 2`] = `
<div>
  Loading
</div>
`;
